Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionWindow/MotionExtractionWindow.cpp

```diff
@@ -463,10 +465,23 @@ namespace EMStudio
     {
         const CommandSystem::SelectionList& selectionList = GetCommandManager()->GetCurrentSelection();
         const size_t numSelectedMotions = selectionList.GetNumSelectedMotions();
+        EMotionFX::RootMotionExtractionData* firstMotionRootExtractionData = nullptr;
 
-        if (numSelectedMotions == 1)
+        for (size_t motionIndex = 0; motionIndex < numSelectedMotions; ++motionIndex)
```

Comment:
umm could this API be improved?

Reply:
Do you mean this loop? It's just looping over the selected motions and use the first motion's extraction data for every other motion. What improvement idea or concern you had in mind?

Reply:
it can probably just return a slice. I think its just an array right?

Reply:
Not sure what you mean, every motion in the selection list need to be saved. Since we are only reflecting the first selected motion into the reflected UI Widget, every motion that selected should get override with the motion extraction value from the first motion as it stored in the UI.
I will update the code to make it more apparent.